### PR TITLE
docs:Updated bpf documentation files

### PR DESCRIPTION
**Pull request by AdityaK60**
Grammatically updated the documentations of bpf.

Signed-off-by: Aditya Kumar <aditya.kumar60@infosys.com>

```release-note
docs: Fix typos in BPF documentation
```

### CODE REVIEW

File: Documentation/bpf/toolchain.rst [hunk 1]
```diff
@@ -1864,7 +1864,7 @@ of all details, but enough for getting started.
     # tc qdisc del dev em1 clsact
 
   tc BPF programs can also be offloaded if the NIC and driver has support for it
-  similarly as with XDP BPF programs. Netronome's nfp supported NICs offer both
+  like XDP BPF programs. Netronome's nfp supported NICs offer both
```
Comment (qmonnet) [inline Documentation/bpf/toolchain.rst:1867]: Missing comma before “like”?

File: Documentation/bpf/toolchain.rst [hunk 2]
```diff
@@ -1775,7 +1775,7 @@ of all details, but enough for getting started.
   It basically means that the ``bpf`` classifier does not need to call into external
   tc action modules, which are not necessary for ``bpf`` anyway, since all packet
   mangling, forwarding or other kind of actions can already be performed inside
-  the single BPF program which is to be attached, and is therefore significantly
+  the single BPF program, and is therefore significantly
```
Comment (qmonnet) [inline Documentation/bpf/toolchain.rst]: “a single BPF program” maybe?

File: Documentation/bpf/toolchain.rst [hunk 3]
```diff
@@ -629,7 +629,7 @@ A full command line example with llc's ``-mcpu=probe``:
     $ llc xdp-example.bc -march=bpf -mcpu=probe -filetype=obj -o xdp-example.o
 
 Generally, LLVM IR generation is architecture independent. There are
-however a few differences when using ``clang --target=bpf`` versus
+however, a few differences when using ``clang --target=bpf`` versus
```
Comment (qmonnet) [inline Documentation/bpf/toolchain.rst]: “There are however,” still sounds strange to me. With another comma maybe: “There are, however,” or even “However/Nonetheless, there are ...”.

**Review by qmonnet — APPROVED** [review]
Thanks! Please find a few remaining nits.

**Review by qmonnet — CHANGES_REQUESTED** [review]
(I meant to request changes)

File: Documentation/bpf/toolchain.rst (hunk 1 above)
Comment (AdityaK60) [reply Documentation/bpf/toolchain.rst:1867]: Ok

File: Documentation/bpf/toolchain.rst [hunk 4]
```diff
@@ -628,8 +628,8 @@ A full command line example with llc's ``-mcpu=probe``:
     $ clang -O2 -Wall --target=bpf -emit-llvm -c xdp-example.c -o xdp-example.bc
     $ llc xdp-example.bc -march=bpf -mcpu=probe -filetype=obj -o xdp-example.o
 
-Generally, LLVM IR generation is architecture independent. There are
-however a few differences when using ``clang --target=bpf`` versus
+Generally, LLVM IR generation is architecture independent.However, there are
```
Comment (qmonnet) [inline Documentation/bpf/toolchain.rst]: Missing space.

```suggestion
Generally, LLVM IR generation is architecture independent. However, there are 
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Looks better, thanks! One missing space to fix, see below.

Could you also squash your commits together into a single commit, please?

**Comment by AdityaK60** [comment]
> Could you also squash your commits together into a single commit, please?

How can I do that? Haven't done it before can you please suggest?

**Comment by qmonnet** [comment]
> How can I do that? Haven't done it before can you please suggest?

Sure! The way I usually do this is in console, with interactive rebases in Git:

1. `git rebase -i main update_bpf_docs` (See the `--interactive` option in `man git rebase` for details)
2. This will open your default `$EDITOR` and show you a list of commits, where you can interactively decide what to do with each commits (available commands are described in the comments below the list):
   
   image

3. Change the actions for the second and third commits to `squash` (or just `s`) if you want to keep each commit title/description, or `fixup` (or just `f`) to discard it. Here we're OK with just the title and description of the first commit, so we can use `f`:

   image

4. Save the file and exit the editor. Git will apply the changes, and in this case squash the three commits together. Check with `git show`.

5. Force-push your changes: `git push -f AdityaK60 update_bpf_docs` or whatever's the name of your remote repository in your local Git setup.

**Review by qmonnet — APPROVED** [review]
Looks all good, thank you!

**Pull request merged**